discovery/mock_test.go

```diff
@@ -15,6 +15,11 @@ import (
 // mockPeer implements the lnpeer.Peer interface and is used to test the
 // gossiper's interaction with peers.
 type mockPeer struct {
+	// embed the peer interface so that mockPeer will implement lnpeer.Peer by
+	// default. Methods that need to be called in tests still need to be
+	// implemented.
+	lnpeer.Peer
```

Comment:
My argument for using this is that it results in tests failing loudly. If you extend an interface and a test uses the new function call it will panic, which you'll pick up and have to go see exactly what your change is affecting. If the tests need the new function, you will have to update the mock in your PR (as before), this just saves on some boilerplate :)

Reply:
I like this boilerplate-free world :)